aspnetcore/fundamentals/middleware/index.md

```diff
@@ -192,13 +192,15 @@ ASP.NET Core ships with the following middleware components, as well as a descri
 | [CORS](xref:security/cors) | Configures Cross-Origin Resource Sharing. | Before components that use CORS. |
 | [Diagnostics](xref:fundamentals/error-handling) | Configures diagnostics. | Before components that generate errors. |
 | [ForwardedHeaders/HttpOverrides](/dotnet/api/microsoft.aspnetcore.builder.forwardedheadersextensions) | Forwards proxied headers onto the current request. | Before components that consume the updated fields (examples: Scheme, Host, ClientIP, Method). |
+| [HTTPS Redirection Middleware](xref:security/enforcing-ssl#require-https) | Redirect all HTTP requests to HTTPS (ASP.NET Core 2.1 or later). | Before components that consume the URL. |
+| [HTTP Strict Transport Security (HSTS) Middleware](xref:security/enforcing-ssl#http-strict-transport-security-protocol-hsts) | Security enhancement middleware (ASP.NET Core 2.1 or later). | Before components that consume the URL. When called with HTTPS Redirection Middleware, call HSTS Middleware before [UseHttpsRedirection](/dotnet/api/microsoft.aspnetcore.builder.httpspolicybuilderextensions.usehttpsredirection). |
```

Comment:
HSTS only adds response headers for requests that are already HTTPS. It is not order sensitive with UseHttpsRedirection. It should go after any request modifications (ForwardedHeaders, rewrite) and before any responses are sent.